dockerfiles/tasks.py

```diff
@@ -14,8 +15,32 @@
 })
 def build(c, cache=False):
     """Build docker image for servers."""
-    cache_opt = '' if cache else '--no-cache'
-    c.run(f'{DOCKER_COMPOSE_COMMAND} build {cache_opt}', pty=True)
+    cache_opt = '--no-cache'
+    cache_hash = ""
+    build_arg = ""
+    if cache:
+        cache_opt = ""
+        files_to_cache = [
+            # Community
+            "requirements/docker.txt",
+
+            # Corporate
+            "../readthedocs.org/requirements/docker.txt",
+            "setup.cfg",
+
+            # Both
+            "../ext-theme/setup.cfg",
+            "../readthedocs-ext/setup.cfg",
+        ]
+
+        for f in files_to_cache:
+            if os.path.exists(f):
+                cache_hash += hashlib.md5(open(f, mode="rb").read()).hexdigest()
```

Comment:
> This would allow `PRUNE_PACKAGE_CACHE` to be set at the host via the same env var. This task could set the env var if it's missing though?

Are you suggesting to remove the hash calculation from this script? Where the value of this variable will be calculated?

Reply:
I updated the code to pass it as environment variable instead of `--build-arg` argument.